src/engine/strat_engine/setup.rs

```diff
@@ -206,6 +208,20 @@ pub fn get_blockdevs(pool_save: &PoolSave, devnodes: &[PathBuf]) -> EngineResult
         let actual_size = try!(blkdev_size(&try!(OpenOptions::new().read(true).open(dev))))
             .sectors();
 
+        match actual_size.cmp(&bda.dev_size()) {
```

Comment:
ah yes I remember seeing some work you did on this, but I'm only seeing Blockdev::recorded_size in master branch. Where'd it go?